pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coredatamodel
+
+import (
+	"errors"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+
+	networking "istio.io/api/networking/v1alpha3"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config"
+	"istio.io/pkg/log"
+)
+
+var notReadyEndpointkey = "networking.alpha.istio.io/notReadyEndpoints"
+
+// Discovery implements all the necessary interfaces
+// to satisfy the aggregated controller, it is used to
+// expose notReadyEndPoints that are received via MCP
+type Discovery interface {
+	RegisterNotReadyEndpoints(conf *model.Config)
+	model.Controller
+	model.ServiceDiscovery
+}
+
+// MCPDiscovery provides storage for NotReadyEndpoints
+type MCPDiscovery struct {
```

Comment:
Does this type need to be public?

Reply:
Yes, it's referenced in bootstrap server for registration of serviceRegistry.

Reply:
OK, then let's get rid of the interface and just call this `Discovery`

Also, should this code live somewhere else? Maybe somewhere under `pilot/pkg/serviceregistry`?

Reply:
Sure, I have removed the interface. As for the placement of discovery I think it makes sense to keep it under coredatamodel (at least for now) since all the MCP related work is in this package.